StripePayments/StripePayments/Source/Helpers/STPBankAccountCollector.swift

```diff
@@ -567,3 +593,93 @@ public class STPBankAccountCollector: NSObject {
         }
     }
 }
+
+// MARK: - Error
+extension STPBankAccountCollector {
+    private func error(
+        for errorCode: STPCollectBankAccountError,
+        loggingSafeErrorMessage: String? = nil
+    ) -> NSError {
+        var userInfo: [String: String] = [:]
+        switch errorCode {
+        case .financialConnectionsSDKNotLinked:
+            userInfo[STPError.errorMessageKey] =
+                "StripeFinancialConnections SDK has not been linked into your project"
+        case .invalidClientSecret:
+            userInfo[STPError.errorMessageKey] = "Unable to parse client secret"
+        case .unexpectedError:
+            userInfo[STPError.errorMessageKey] = loggingSafeErrorMessage
+        }
+        return STPBankAccountCollectorError(code: errorCode, loggingSafeUserInfo: userInfo) as NSError
+    }
+}
+
+/// STPBankAccountCollector errors (i.e. errors that are created by the STPBankAccountCollector class and have a corresponding STPCollectBankAccountError) used to be NSErrors.
+/// This struct exists so that these errors can be Swift errors to conform to AnalyticLoggableError, while still looking like the old NSErrors to users (i.e. same domain and code).
+struct STPBankAccountCollectorError: Error, CustomNSError, AnalyticLoggableError {
+    // AnalyticLoggableError properties
+    let analyticsErrorType: String = errorDomain
+    let analyticsErrorCode: String
+    let additionalNonPIIErrorDetails: [String: Any]
+
+    // CustomNSError properties, to not break old behavior when this was an NSError
+    static let errorDomain: String = "STPBankAccountCollectorErrorDomain"
+    let errorUserInfo: [String: Any]
+    let errorCode: Int
```

Comment:
I understand that errorDomain, errorUserInfo, and errorCode are adding conformance to CustomNSError, but it seems that I can comment out these and still build? I was expecting this not to build somehow?